vignettes/generate-lockfile.Rmd

```diff
@@ -0,0 +1,71 @@
+---
+title: "Generate renv lockfile from log file"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{generate-lockfile}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(logrx)
+library(stringr)
+library(dplyr)
+library(renv)
+```
+
+Axecute a script and get the log file.
+
+```{r run_axecute, echo = FALSE}
+scriptPath <- tempfile()
+
+logDir <- tempdir()
+writeLines(
+  c(
+    "print('hello logrx')", 
+    "library(dplyr)",
+    "count(iris, Species)"
+  ),
+  con = scriptPath
+)
+
+filePath <- file.path(logDir, "log_out_parse")
+
+axecute(scriptPath, log_name = "log_out_parse", log_path = logDir)
+```
+
+Later on read (and parse) previous log file as list of objects. 
+
+```{r read_logfile}
+parsedFile <- read_log_file(filePath)
+```
+
+Create a renv lockfile based on the used packages mentioned in the log file.
+
+```{r gen_lockfile}
+used_pgks <- parsedFile$`Used Package and Functions` %>%
+  transmute(package_name = str_extract(library, "(?<=package\\:).+"))
+
+temp_lockfile <- tempfile(fileext = ".lockfile")
+
+x <- renv::lockfile_create(
+  type = "custom",
+  packages = used_pgks$package_name
+)
+
+renv::lockfile_write(
```

Comment:
@parmsam-pfizer might be excessive but will this just record packages with the latest version or also the versions used?  wasn't sure if that was even doable with this method or if we'd have to go all the way to an install and then lockfile create approach to capture that information.